pkg/bpf/bpffs.go

```diff
@@ -104,100 +123,226 @@ func OpenAfterMount(m *Map) error {
 	return nil
 }
 
+// removeMap removes a map from the BPF filesystem.
+func removeMap(name string) {
+	os.Remove(MapPath(name))
```

Comment:
If this code is really necessary, it might be good to add a comment highlighting that we ignore the error here because it is a best-effort deletion handling upgrade cases to ensure that old unused maps are cleaned up.